fxa-oauth-server/lib/routes/token.js

```diff
@@ -334,6 +358,23 @@ async function validateRefreshTokenGrant(client, params) {
   return tokObj;
 }
 
+
+async function validateAssertionGrant(client, params) {
```

Comment:
nit: we might find it useful to extract the functions below the route handler into their own files / modules. In case we need more testing or just cleaner for the future